src/pages/Forever/useRedeemCode.ts

```diff
@@ -0,0 +1,81 @@
+import { useCurrentLocale } from 'components/utils/CurrentLocale'
+import { useRedeemCodeV2Mutation } from 'data/graphql'
+import { FormikHelpers } from 'formik'
+import { useHistory } from 'react-router'
+import { captureSentryError } from 'utils/sentry-client'
+import { RedeemCodeFormValue } from './components/RedeemCode'
+
+export const useRedeemCode = () => {
+  const [redeemCode] = useRedeemCodeV2Mutation()
```

Comment:
what about also returning the loading state from here? You could use `const [redeemCode, { loading }] = useRedeemCodeV2Mutation()` or something. Instead of creating our own loading state

Reply:
I used the built in `isSubmitting` from Formik, but we could totally use that if it seems off

Reply:
Aah sorry i missed that. I guess thats ok!